pkg/controllers/status/status_test.go

```diff
@@ -0,0 +1,201 @@
+package status
+
+import (
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"testing"
+	"time"
+)
+
+const (
+	clusterRolesExists        = "ClusterRolesExists"
+	clusterRoleBindingsExists = "ClusterRoleBindingsExists"
+	failedToCreateRoles       = "FailedToCreateRoles"
+)
+
+func TestAddCondition(t *testing.T) {
+	mockTime := time.Unix(0, 0)
+	mockErr := errors.New("mock error")
+
+	tests := map[string]struct {
+		conditions     []v1.Condition
+		condition      v1.Condition
+		reason         string
+		err            error
+		wantConditions []v1.Condition
+	}{
+		"add new condition": {
+			conditions: []v1.Condition{},
+			condition:  v1.Condition{Type: clusterRolesExists},
+			reason:     clusterRolesExists,
+			err:        nil,
+			wantConditions: []v1.Condition{
+				{
+					Type:   clusterRolesExists,
+					Status: v1.ConditionTrue,
+					Reason: clusterRolesExists,
+					LastTransitionTime: v1.Time{
+						Time: mockTime,
+					},
+				},
+			},
+		},
+		"add new condition when there are already other existing conditions": {
+			conditions: []v1.Condition{
+				{
+					Type:   clusterRolesExists,
+					Status: v1.ConditionTrue,
+					Reason: clusterRolesExists,
+					LastTransitionTime: v1.Time{
+						Time: mockTime,
+					},
+				},
+			},
+			condition: v1.Condition{Type: clusterRoleBindingsExists},
+			reason:    clusterRoleBindingsExists,
+			err:       nil,
+			wantConditions: []v1.Condition{
+				{
+					Type:   clusterRolesExists,
+					Status: v1.ConditionTrue,
+					Reason: clusterRolesExists,
+					LastTransitionTime: v1.Time{
+						Time: mockTime,
+					},
+				},
+				{
+					Type:   clusterRoleBindingsExists,
+					Status: v1.ConditionTrue,
+					Reason: clusterRoleBindingsExists,
+					LastTransitionTime: v1.Time{
+						Time: mockTime,
+					},
+				},
+			},
+		},
+		"add new condition with error": {
+			conditions: nil,
+			condition:  v1.Condition{Type: clusterRolesExists},
+			reason:     failedToCreateRoles,
+			err:        mockErr,
+			wantConditions: []v1.Condition{
+				{
+					Type:    clusterRolesExists,
+					Status:  v1.ConditionFalse,
+					Message: mockErr.Error(),
+					Reason:  failedToCreateRoles,
+					LastTransitionTime: v1.Time{
+						Time: mockTime,
+					},
+				},
+			},
+		},
+		"modify existing condition": {
+			conditions: []v1.Condition{
+				{
+					Type:   clusterRolesExists,
+					Status: v1.ConditionTrue,
+					Reason: clusterRolesExists,
+					LastTransitionTime: v1.Time{
+						Time: mockTime,
```

Comment:
For this one, could you set the time here to something else so that we the `LastTransitionTime` gets updated?

Reply:
Actually I missed to update the `LastTransitionTime` when a condition gets updated. Fixed it and added a tests in https://github.com/rancher/rancher/pull/47407/commits/2be7072141c8f604cde5df1433115524ba6f3b1d 
Thanks for catching this!